shoop/core/models/_orders.py

```diff
@@ -345,6 +345,9 @@ def save(self, *args, **kwargs):
         super(Order, self).save(*args, **kwargs)
         if first_save:  # Have to do a double save the first time around to be able to save identifiers
             self._save_identifiers()
+        for line in self.lines.all():
+            if line.product_id:
```

Comment:
Do this filtering during the queryset: `for line in self.lines.exclude(product_id=None)`